interchaintest/fee_middleware_test.go

```diff
@@ -0,0 +1,199 @@
+	client, network := interchaintest.DockerSetup(t)
+
+	rf := relayertest.NewRelayerFactory(relayertest.RelayerConfig{InitialBlockHistory: 50})
+	r := rf.Build(t, client, network)
+
+	const pathChainAChainB = "chainA-chainB"
+
+	// Build the network
+	ic := interchaintest.NewInterchain().
+		AddChain(chainA).
+		AddChain(chainB).
+		AddRelayer(r, "relayer").
+		AddLink(interchaintest.InterchainLink{
+			Chain1:  chainA,
+			Chain2:  chainB,
+			Relayer: r,
+			Path:    pathChainAChainB,
+			CreateChannelOpts: ibc.CreateChannelOptions{
+				SourcePortName: "transfer",
+				DestPortName:   "transfer",
+				Order:          ibc.Unordered,
+				Version:        "{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}",
+			},
+			CreateClientOpts: ibc.DefaultClientOpts(),
+		})
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	require.NoError(t, ic.Build(ctx, eRep, interchaintest.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: false,
+	}))
+
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	err = testutil.WaitForBlocks(ctx, 10, chainA, chainB)
+	require.NoError(t, err)
+
+	// ChainID of ChainA
+	chainID_A := chainA.Config().ChainID
+
+	// Channel of ChainA
+	chA, err := r.GetChannels(ctx, eRep, chainID_A)
+	require.NoError(t, err)
+	channelA := chA[0]
+
+	// Fund a user account on chain1 and chain2
+	const userFunds = int64(1_000_000_000_000)
+	users := interchaintest.GetAndFundTestUsers(t, ctx, t.Name(), userFunds, chainA, chainB)
+	walletA := users[0]
+	userAddressA := walletA.FormattedAddress()
+	walletB := users[1]
+	userAddressB := walletB.FormattedAddress()
+
+	// Addresses of both the chains
+	WalletA, _ := r.GetWallet(chainA.Config().ChainID)
+	rlyAddressA := WalletA.FormattedAddress()
+
+	WalletB, _ := r.GetWallet(chainB.Config().ChainID)
+	rlyAddressB := WalletB.FormattedAddress()
+
+	// register CounterpartyPayee
+	cmd := []string{
+		"tx", "register-counterparty",
+		chainA.Config().Name,
+		channelA.ChannelID,
+		"transfer",
+		rlyAddressA,
+		rlyAddressB,
+	}
+	_ = r.Exec(ctx, eRep, cmd, nil)
+	require.NoError(t, err)
+
+	// Query the relayer CounterPartyPayee on a given channel
+	query := []string{
+		chainA.Config().Bin, "query", "ibc-fee", "counterparty-payee", channelA.ChannelID, rlyAddressA,
+		"--chain-id", chainID_A,
+		"--node", chainA.GetRPCAddress(),
+		"--home", chainA.HomeDir(),
+		"--trace",
+	}
+	_, _, err = chainA.Exec(ctx, query, nil)
+	require.NoError(t, err)
+
+	// Get initial account balances
+	userAOrigBal, err := chainA.GetBalance(ctx, userAddressA, chainA.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userFunds, userAOrigBal)
+
+	userBOrigBal, err := chainB.GetBalance(ctx, userAddressB, chainB.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userFunds, userBOrigBal)
+
+	rlyAOrigBal, err := chainA.GetBalance(ctx, rlyAddressA, chainB.Config().Denom)
```

Comment:
You are checking the balance of `rlyAddressA` for the native denom from `ChainB` this works because both chains are using the same denom but this could break if that changes, can we change this to use `chainA.Config().Denom`?